934 project status filter

dmijatovic opened the pull request:
# Project status filter

Closes #933 
Closes #934 
Closes #937

Changes proposed in this pull request:
* Adding project status filter to project overview page. The filter values are: Pending, In progress, Finished and Unknown. We have chosen a single value filter because the project can have only one of the values. The other filters support multiple values in the AND fashion.
* Adding projects status filter to organisation project tab. Filter uses same values as on the project overview page.
* Improved project period presentation on the project card:
   - Project with status `unknown` do not show any period indication
   - Finished projects show period and progress bar when both values are present. When start date is not present we y show period labels: N/S - Jan 2022
   - In progress projects have always start and end date and show both period labels and progress bar
   - Pending projects show progress bar, if both date values are present, and the period labels. In case one of date values is missing it uses N/S (short for not specified)
* Aligned project status on the project page (top right corner of the page) and in the related project cards on software and project page

How to test:
* `make start` to build and generate test data
* navigate to project overview page and test Project status filter in combination with other filters
* clear filters. Selected order should be kept and filters should be removed. The clear button should be disabled.
* navigate to software overview page. Confirm that clear button works in similar way as on project overview page
* The project status in the related projects on the software page should have use same values as the filter (Pending, In progress, Finished)
* navigate to project page. The status at the top right corner should use same status values as the filter (Pending, In progress, Finished)
* navigate to organisation page:
    * The page should open on software tab, default order should be mentions and the clear button is disabled. After selecting some filters clear button should be enabled. Clear button should clear filters but keep selected order.
    * navigate to project tab. Default order should be impact and the clear button is disabled. After selecting some filters clear button should be enabled. Clear button should clear filters but keep selected order.

## Example project overview page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/1e3ed936-4d11-456c-9345-19c795de94f5)

## Example organisation projects page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/15a14075-5a33-4754-964c-43af636ba791)


PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [ ] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

jmaassen commented:
Some comments: 

- the behavior of the "status" dropdown menu is different from the others: when selecting an item from "keywords" for example, the selected value is immutable and shown in a rounded box. The others have the same behavior. In the status pulldown menu however, the selected item is simply text, which I can also change. It would be better is this is also immutable. 
- Also, in the others I can select multiple values, but for status I can use only one? Why can't I select "in progess + pending projects"? (which equals "not finished")
- clearing the filters also changes the sort order. This is a bit unexpected, but could be solved by renaming the button "reset" instead of "clear"?

dmijatovic commented:
@jmaassen and @ewan-escience I changed the behavour of clear button and project status filter. Can you please confirm that this approach is better. After that I will implement same approach in the organisation projects page. Thanks!

BTW: I also rebased this branch and rsd-v2-release with the latest changes from main

jmaassen reviewed (approved):
Looks good. Don't forget to also add it in the org pages?
